### PR TITLE
feat(core): [Stripe] add bank debits payment method to stripe

### LINKED ISSUE #838: [FEATURE] Add support for Bank Debits payment method
### Problem statement
Bank Debits enables merchants to directly pull funds from the customers' bank accounts once the customers provide authorization for the same. It is primarily used for recurring transactions/mandates and for large ticket transactions like rent, fees, etc.
Hyperswitch will need to support the following popular Bank Debits from US and EU:

- ACH Direct Debit
- BECS Direct Debit
- BACS Direct Debit
- SEPA Direct Debit

### Possible Implementation

_No response_

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Yes

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] New feature

## Description

This PR will add support for creating payments using `bank_debits` payment method. This has been implemented for stripe.
Mandate details have to passed for all the payment methods.
- Sample mandate data
```json
{
"mandate_data": {
        "customer_acceptance": {
            "acceptance_type": "online",
            "accepted_at": "2022-09-10T10:11:12Z",
            "online": {
                "ip_address": "123.32.25.123",
                "user_agent": "Mozilla/5.0 (Linux; Android 12; SM-S906N Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/80.0.3987.119 Mobile Safari/537.36"
            }
        },
        "mandate_type": {
            "single_use": {
                "amount": 6540,
                "currency": "USD"
            }
        }
    },
}
```
- Ach Bank debit in US. Create a `merchant_connector_account` with `business_country` as `US`.
```json
{
"currency":"USD",
"business_country": "US",
"payment_method": "bank_debit",
    "payment_method_type": "sepa",
    "payment_method_data": {
        "bank_debit": {
            "ach_bank_debit": {
                "billing_details": {
                    "name": "John Doe",
                    "email": "johndoe@example.com"
                },
                "account_number": "000123456789",
                "routing_number": "110000000"
            }
        }
    }
}
```

- Sepa. Create a `merchant_connector_account` in any country of the EU region, example "FR".
```json
{
"currency":"EUR",
"business_country":"FR",
"payment_method": "bank_debit",
    "payment_method_type": "sepa",
    "payment_method_data": {
        "bank_debit": {
            "sepa_bank_debit": {
                "billing_details": {
                    "name": "John Doe",
                    "email": "johndoe@example.com"
                },
                "iban": "DE89370400440532013000"
            }
        }
    }
}
```

- Becs. Create a `merchant_connector_account` in `AU` region.
```json
{
"currency":"AUD",
"payment_method": "bank_debit",
    "payment_method_type": "becs",
    "payment_method_data": {
        "bank_debit": {
            "becs_bank_debit": {
                "billing_details": {
                    "name": "John Doe",
                    "email": "johndoe@example.com"
                },
                "account_number": "000123456",
                "bsb_number": "000000"
            }
        }
    }
}
```
- Bacs
```json
{
"payment_method_type": "bacs",
    "payment_method_data": {
        "bank_debit": {
            "bacs_bank_debit": {
                "billing_details": {
                    "name": "John Doe",
                    "email": "johndoe@example.com",
                    "address": {
                        "line1": "Street",
                        "line2": "Area",
                        "city": "LDN",
                        "zip": "SW1A",
                        "country": "GB"
                    }
                },
                "account_number": "00012345",
                "sort_code": "108800"
            }
        }
    }
}
```

## Motivation and Context

To support `bank_debits` payment method for anyone to use. Closes #838 

## How did you test it?

- ACH bank debit
Screenshot 2023-04-18 at 12 43 41 PM
- Microdeposit verification for ACH
Screenshot 2023-04-18 at 12 45 26 PM

- Sepa bank debit
Screenshot 2023-04-18 at 12 56 24 PM

- Bacs bank debit
The payment goes to `requires_action`. This requires some options to be enabled in stripe account.

- Becs bank debit
Screenshot 2023-04-19 at 12 50 14 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/stripe/transformers.rs [hunk 1]
```diff
@@ -2,7 +2,10 @@ use std::str::FromStr;
 
 use api_models::{self, enums as api_enums, payments};
 use base64::Engine;
-use common_utils::{fp_utils, pii::Email};
+use common_utils::{
+    fp_utils,
+    pii::{self},
```
Comment (Abhicodes-crypto) [inline crates/router/src/connector/stripe/transformers.rs:7]: u can remove self here

**Review by Abhicodes-crypto — APPROVED** [review]
LGTM

File: crates/router/src/connector/stripe/transformers.rs [hunk 2]
```diff
@@ -201,13 +218,53 @@ pub struct StripeBankRedirectData {
     pub bank_specific_data: Option<BankSpecificData>,
 }
 
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(untagged)]
```
Comment (jarnura) [inline crates/router/src/connector/stripe/transformers.rs:222]: Why we need untagged here the parent struct which use the this have method type we can infer that and serialize.

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Review by jarnura — CHANGES_REQUESTED** [review]
Add required labels

**Review by jarnura — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by Abhicodes-crypto — APPROVED** [review]
LGTM

**Pull request merged**